src/storage/src/hummock/conflict_detector.rs

```diff
@@ -56,17 +56,15 @@ impl ConflictDetector {
     pub fn set_watermark(&self, epoch: HummockEpoch) {
         loop {
             let current_watermark = self.get_epoch_watermark();
-            assert!(
-                epoch > current_watermark,
-                "not allowed to set epoch watermark to equal to or lower than current watermark: current is {}, epoch to set {}",
-                current_watermark,
-                epoch
-            );
+            if epoch <= current_watermark {
```

Comment:
Why do we remove this assert? Is there any scenario in concurrent checkpoint that the watermark to be set can be lower than the current watermark?

Reply:
The watermark to be set can be lower than the current watermark, But this method is only to reduces the size of map.

Reply:
So , if the watermark to be set can be lower than the current watermark , We don't need to do anything.